qrtool: 0.8.5 -> 0.10.0

philiptaron opened the pull request:
## Description of changes

https://github.com/sorairolake/qrtool/blob/develop/CHANGELOG.adoc

I added a [`passthru.updateScript = nix-update-script { };`](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript) so that likely this will be the last hand-rolled update.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

philiptaron commented:
Thanks for the ping that you'd updated this, @sorairolake.

artemist commented:
Working for me on x86_64-linux. I'm not sure why you changed it to not use rec but it doesn't seem to break nix-update so it's probably fine (I can't figure out how to test nixpkgs-update)

artemist commented on pkgs/by-name/qr/qrtool/package.nix:
```diff
@@ -6,18 +6,18 @@
     repo = "qrtool";
-    rev = "v${version}";
-    sha256 = "sha256-jrvNZGO1VIDo6Mz3NKda1C7qZUtF9T00CAFK8yoGWjc=";
+    rev = "v0.9.0";
```
I'm not sure why you made this change but it seems fine

kirillrdy replied:
I am not sure why either, I would revert

kirillrdy replied:
sorry I've now read the reason for removing `rec` may I suggest still adding `let version = "0.9.0";  in rustPlatform` 

until everything is migrated to `finalAtts`

philiptaron replied:
> sorry I've now read the reason for removing rec may I suggest still adding let version = "0.9.0";  in rustPlatform

I did this.

philiptaron replied:
I undid this.

atalii reviewed (commented):
There's no strict need to remove rec, but there's no strict need to keep it either. LGTM pending nixpkgs-review.

philiptaron commented:
I used [nurl](https://github.com/nix-community/nurl) to get the hash, which obviated the one use of a recursive attribute. Hence its removal.

kirillrdy commented on pkgs/by-name/qr/qrtool/package.nix:
```diff
@@ -6,18 +6,18 @@
   };
 
-  cargoSha256 = "sha256-JOnvlabCr3fZsIIRc2qTjf50Ga83zL8Aoo2sqzMBs7g=";
+  cargoSha256 = "sha256-bB07rZ4QZ8ALIQudbJAqudxUFe68LSX6AKxNFy5xg+8=";
```
```suggestion
  cargoHash = "sha256-bB07rZ4QZ8ALIQudbJAqudxUFe68LSX6AKxNFy5xg+8=";
```

philiptaron replied:
https://ryantm.github.io/nixpkgs/languages-frameworks/rust/ doesn't recommend one over the other.

sorairolake commented:
[v0.10.0](https://github.com/sorairolake/qrtool/releases/tag/v0.10.0) has been released.

kirillrdy commented on pkgs/by-name/qr/qrtool/package.nix:
```diff
@@ -6,37 +6,42 @@
   postInstall = ''
     # Built by ./build.rs using `asciidoctor`
-    installManPage ./target/*/release/build/qrtool*/out/*.?
+    installManPage ./target/*/release/build/${name}*/out/*.?
```
using `pname` in here and other places is an anti pattern.

eg if repo gets renamed tomorrow ( eg fork ) unlikely that all other references will also be renamed

philiptaron commented:
That’s how I feel about the version reference you asked me to keep!

Frankly I have a bit of whiplash. Either constants like this ought to be
extracted or they shouldn’t. One rule shouldn’t apply to name and another
to version.

I prefer the “duplications” patch with version and an explicitly specified
tag that I originally sent out.



On Sun, Dec 17, 2023 at 10:46 AM kirillrdy ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In pkgs/by-name/qr/qrtool/package.nix
> <https://github.com/NixOS/nixpkgs/pull/274806#discussion_r1429240878>:
>
> >
>    nativeBuildInputs = [ asciidoctor installShellFiles ];
>
>    postInstall = ''
>      # Built by ./build.rs using `asciidoctor`
> -    installManPage ./target/*/release/build/qrtool*/out/*.?
> +    installManPage ./target/*/release/build/${name}*/out/*.?
>
> using pname in here and other places is an anti pattern.
>
> eg if repo gets renamed tomorrow ( eg fork ) unlikely that all other
> references will also be renamed
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/274806#pullrequestreview-1785495028>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAKWV47OG3PQIN3GOLI5ILYJ442FAVCNFSM6AAAAABAXZNAIWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOOBVGQ4TKMBSHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kirillrdy commented:
> Frankly I have a bit of whiplash. Either constants like this ought to be extracted or they shouldn’t. One rule shouldn’t apply to name and another to version.

`version` will be changed during next version update, and it will always be changed everywhere that's why it makes sense for it to be a variable.

`pname`  and other references are coincidental. eg if repo gets forked, all build instructions will need to be pointing again.

as per previous comment, `There's no strict need to remove rec`, original version of master doesn't have those issues

philiptaron commented:
> version will be changed during next version update, and it will always be changed everywhere that's why it makes sense for it to be a variable.

`version` will always be changed, but it's also coincidental -- and more likely to change -- in its use in the `rev` parameter. @sorairolake happens to tag releases in this pattern. If they switched to `release-<version>` next release, or skip the `v` prefix -- just as in the case with the fork! -- the pattern would need to be changed.

It seems fair to me to encode the current name and structure.

As the "maintainer" of this module, this back-and-forth (which happened on the last one!) makes me regret not taking a harder line that I'm going to just copy the tag that's on GitHub and be done, rather than "reverting" to either the `rec` version or making a `let` version that unites the `rev` argument and the `version` argument.

philiptaron committed:
qrtool: add `nix-update-script`

See [the documentation for more](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript).

kirillrdy commented on pkgs/by-name/qr/qrtool/package.nix:
```diff
@@ -4,20 +4,21 @@
     repo = "qrtool";
     rev = "v${version}";
-    sha256 = "sha256-jrvNZGO1VIDo6Mz3NKda1C7qZUtF9T00CAFK8yoGWjc=";
+    sha256 = "sha256-PXjggeDofrrS32XdqnOgAMlAiw+/7eMQpELw8EmiwNI=";
```
```suggestion
    hash = "sha256-PXjggeDofrrS32XdqnOgAMlAiw+/7eMQpELw8EmiwNI=";
```

philiptaron replied:
I think I'll leave them. There's no guidance one way or the other that I can find in the docs and this way works just fine.

kirillrdy replied:
https://nixos.org/manual/nixpkgs/stable/#compiling-rust-applications-with-cargo

```
buildRustPackage requires either the cargoSha256 or the cargoHash attribute which is computed over all crate sources of this package. cargoHash256 is used for traditional Nix SHA-256 hashes, such as the one in the example above. cargoHash should instead be used for [SRI](https://www.w3.org/TR/SRI/) hashes. For example:
```

same logic applies for `sha256`

kirillrdy commented on pkgs/by-name/qr/qrtool/package.nix:
```diff
@@ -4,20 +4,21 @@
   };
 
-  cargoSha256 = "sha256-JOnvlabCr3fZsIIRc2qTjf50Ga83zL8Aoo2sqzMBs7g=";
+  cargoSha256 = "sha256-0Vwx80XxrJX8C293+4qgRr0pTMV6Jy3IBNQISzFxyr0=";
```
```suggestion
  cargoHash = "sha256-0Vwx80XxrJX8C293+4qgRr0pTMV6Jy3IBNQISzFxyr0=";
```

kirillrdy reviewed (commented):
I appreciate your patience :heart:

kirillrdy commented:
@philiptaron can you squash last two commits and change commit message to match https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

philiptaron commented:
Sorry, I'm on my phone without access to a real computer. I clicked the "accept suggestion" button on GitHub. If you've got the right to commit, I believe you can also squash the commits together.

kirillrdy commented:
> Sorry, I'm on my phone without access to a real computer. I clicked the "accept suggestion" button on GitHub. If you've got the right to commit, I believe you can also squash the commits together.

no rush

philiptaron committed:
qrtool: 0.8.5 -> 0.10.0

Also, update `sha256` to `hash` and `cargoSha256` to `cargoHash` per feedback from @kirillrdy.

Co-authored-by: kirillrdy <kirillrdy@gmail.com>

kirillrdy reviewed (approved):
Result of `nixpkgs-review pr 274806` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qrtool</li>
  </ul>
</details>

atalii reviewed (approved):
This all looks and works good; thanks for your patience :)

philiptaron commented:
Obviated by 
* https://github.com/NixOS/nixpkgs/pull/276188

kirillrdy commented:
interesting ... so qrtool still doesn't have update-script but auto updated was still generated. If anyone knows exactly how that worked I would like to know
